pkg/controllers/provisioningv2/rke2/provisioningcluster/template.go

```diff
@@ -303,6 +303,10 @@ func machineDeployments(cluster *rancherv1.Cluster, capiCluster *capi.Cluster, d
 		if !machinePool.DrainBeforeDelete {
 			machineSpecAnnotations[capi.ExcludeNodeDrainingAnnotation] = "true"
 		}
+		if machinePool.EtcdRole {
+			logrus.Warnf("Ignoring drain before delete for pool [%s] with etcd role", machinePoolName)
```

Comment:
This message will be printed a bunch (not just on create). I think a comment here is sufficient. Maybe even combine this if-statement with the one above.

This behavior is something that should go in the docs and be release noted.